crates/cairo-addons/src/vm/hints.rs

```diff
@@ -79,11 +91,82 @@ impl HintProcessor {
         Self {
             inner: BuiltinHintProcessor::new(self.inner.extra_hints, run_resources),
             python_hints: self.python_hints,
+            dynamic_hint_executor: self.dynamic_hint_executor,
         }
     }
 
-    pub fn build(self) -> BuiltinHintProcessor {
-        self.inner
+    /// Add support for dynamic Python hints
+    #[must_use]
+    pub fn with_dynamic_python_hints(mut self) -> Self {
+        // Store the generic Python hint executor for fallback
+        self.dynamic_hint_executor = Some(generic_python_hint().func.clone());
+        self
+    }
+
+    /// Build the hint processor
+    pub fn build(self) -> HintProcessor {
+        HintProcessor {
+            inner: self.inner,
+            python_hints: self.python_hints,
+            dynamic_hint_executor: self.dynamic_hint_executor,
+        }
+    }
+}
+
+impl HintProcessorLogic for HintProcessor {
+    /// Executes a hint. If the hint is not found, it will try to execute the hint as Python code
+    /// using the dynamic hint executor.
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn std::any::Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<(), HintError> {
+        // Try to execute the hint with the inner processor
+        let result = self.inner.execute_hint(vm, exec_scopes, hint_data, constants);
+
+        match result {
+            Ok(_) => Ok(()),
+            Err(HintError::UnknownHint(_)) => {
```

Comment:
An explicit warning should be thrown to ensure the dev is knowing the dynamic_hint_executor is used and it is debug only.